release/ray_release/test_automation/state_machine.py

```diff
@@ -0,0 +1,50 @@
+from ray_release.test import (
+    Test,
+    TestState,
+)
+
+
+class TestStateMachine:
```

Comment:
could you add a comment of what this state machine does at the high level?

like what is the end-goal logic that this is trying to implement?